tests/integration/test_network_configuration.py

```diff
@@ -0,0 +1,248 @@
+external breaking behaviour. In the future we can update this test suite to
+correspond to the behaviour we want, and we get a todo list of things to
+change 😂
+"""
+import json
+
+import pytest
+import requests
+import xmltodict
+from botocore.auth import SigV4Auth
+
+from localstack import config
+from localstack.aws.api.lambda_ import Runtime
+from localstack.testing.aws.lambda_utils import is_new_provider, is_old_provider
+from localstack.utils.files import new_tmp_file, save_file
+from localstack.utils.strings import short_uid
+
+# TODO: how do we test `localstack_hostname` - this variable configures the
+# host that services make requests to when starting up (e.g. opensearch) and
+# they won't start if we override the variable.
+
+
+pytestmark = [pytest.mark.only_localstack]
+
+
+class TestOpenSearch:
+    """
+    OpenSearch does not respect any customisations and just returns a domain with localhost.localstack.cloud in.
+    """
+
+    def test_default_strategy(
+        self, opensearch_client, opensearch_wait_for_cluster, assert_host_customisation
+    ):
+        domain_name = f"domain-{short_uid()}"
+        res = opensearch_client.create_domain(DomainName=domain_name)
+        opensearch_wait_for_cluster(domain_name)
+        endpoint = res["DomainStatus"]["Endpoint"]
+
+        assert_host_customisation(endpoint, use_localstack_cloud=True)
+
+    def test_port_strategy(
+        self, monkeypatch, opensearch_client, opensearch_wait_for_cluster, assert_host_customisation
+    ):
+        monkeypatch.setattr(config, "OPENSEARCH_ENDPOINT_STRATEGY", "port")
+
+        domain_name = f"domain-{short_uid()}"
+        res = opensearch_client.create_domain(DomainName=domain_name)
+        opensearch_wait_for_cluster(domain_name)
+        endpoint = res["DomainStatus"]["Endpoint"]
+
+        if config.is_in_docker:
+            assert_host_customisation(endpoint, use_localhost=True)
+        else:
+            assert_host_customisation(endpoint, custom_host="127.0.0.1")
+
+    def test_path_strategy(
+        self, monkeypatch, opensearch_client, opensearch_wait_for_cluster, assert_host_customisation
+    ):
+        monkeypatch.setattr(config, "OPENSEARCH_ENDPOINT_STRATEGY", "path")
+
+        domain_name = f"domain-{short_uid()}"
+        res = opensearch_client.create_domain(DomainName=domain_name)
+        opensearch_wait_for_cluster(domain_name)
+        endpoint = res["DomainStatus"]["Endpoint"]
+
+        assert_host_customisation(endpoint, use_localstack_hostname=True)
+
+
+class TestS3:
+    @pytest.mark.skipif(
+        condition=config.LEGACY_S3_PROVIDER, reason="Not implemented for legacy provider"
+    )
+    def test_non_us_east_1_location(
+        self, monkeypatch, s3_resource, s3_client, cleanups, assert_host_customisation
+    ):
+        monkeypatch.setattr(config, "LEGACY_S3_PROVIDER", False)
+        monkeypatch.setenv("PROVIDER_OVERRIDE_S3", "asf")
```

Comment:
This won't work I believe, as the provider will be loaded when one call is going to S3, which might be before this test. I suppose this test will be running only once the new ASF provider will be default. We could also add it manually to the S3 test suite, because we have a custom CI job running the S3 test suite with `PROVIDER_OVERRIDE_S3=asf`. Or I don't know if we can add these tests to this CI job? Maybe @alexrashed could help us here.

Reply:
That's true, providers cannot be changed at runtime once they have been loaded once.
In order to add this test to be executed in the S3 ASF provider tests in CircleCI it has to be added to the test path in the workflow: https://github.com/localstack/localstack/blob/204e605e6fc47331718201724cb2732179ecb772/.circleci/config.yml#L175

Reply:
Thanks. I think I left that line in before I set the test to be skipped if using the legacy provider. Thanks for the tip @alexrashed I can add `tests/integration/test_network_configuration.py::TestS3` to this line and the network tests will run with the new provider, thanks!

Reply:
I'll do the same with the lambda tests as well